File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    // 뒤에있는 기능이 먼저되면 앞에있는 기능이 완료될때 같이 배포됨
    // 100-원소 / speeds[i] > [7,3,9] 로 만들고 (Math.ceil() 올림 사용)
    // count = 0에서 1씩 더하고 다음요소가 크면 []인 answer에 넣고 리턴
    let answer = [];
    let proSpeed = [];
    let count = 0;
    for(let i = 0; i< progresses.length; i++){
        proSpeed.push(Math.ceil((100-progresses[i])/speeds[i]))
    }
    
    while(proSpeed.length > 0) {
        let cur = proSpeed[0];
        let count = 0;

        while(proSpeed[0] <= cur) {
            proSpeed.shift();
            count++;
        }
        
        answer.push(count);
    }
    
    return answer
    
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.ArrayList;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> queue = new LinkedList<>();
        for(int i=0; i<progresses.length; i++) {
            queue.add((100-progresses[i]) % speeds[i] > 0 ? (100-progresses[i]) / speeds[i] + 1 : (100-progresses[i]) / speeds[i]);
        }
        
        ArrayList<Integer> answerList = new ArrayList<>();
        int preWorkDay = queue.poll();
        int numOfSameWorkDay = 1;
        while(!queue.isEmpty()) {
            int workDay = queue.poll();
            if(preWorkDay >= workDay) {
                numOfSameWorkDay++;
            } else {
                answerList.add(numOfSameWorkDay);
                numOfSameWorkDay = 1;
                preWorkDay = workDay;
            }
        }
        answerList.add(numOfSameWorkDay);
        
        int[] answer = new int[answerList.size()];
        for(int i=0; i<answer.length; i++) {
            answer[i] = answerList.get(i);
        }
        return answer;
    }
}